crates/testing/tests/da_task.rs

```diff
@@ -80,7 +87,20 @@ async fn test_da_task() {
             output.insert(SequencingHotShotEvent::DAVoteSend(vote), 1);
         }
     }
+    output.insert(
+        SequencingHotShotEvent::VidDisperseSend(
+            VidDisperse {
+                view_number: message.data.view_number,
+                shares,
+                common,
+            },
+            pub_key,
+        ),
+        1,
+    );
+    // TODO why does this test pass without a `VidVoteSend` event here? https://github.com/EspressoSystems/HotShot/issues/1697
```

Comment:
Have you tried adding `VidVoteSend` to `output`? This test terminates once all the expected outputs are received, so maybe it terminates before getting `VidVoteSend`, which isn't in `output`.

Reply:
I tried adding `VidVoteSend` to `output`. The test hangs when I do that. Seems that there's something wrong in the way I've copied the VID vote code from DA vote.

Reply:
Solved. I need to add `VidDisperseRecv` to both `input` and `output`. Then the test no longer hangs when I add `VidVoteSend` to `output`.